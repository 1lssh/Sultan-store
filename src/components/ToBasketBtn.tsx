import { FC } from "react";
import cardBasket from '../source/icons/card-basket.png'

export const ToBasketBtn: FC = () => {
	return (
		<div className='good-card__button'>
			<button>В корзину <img src={cardBasket} alt="basket" /></button>
		</div>
	);
}
