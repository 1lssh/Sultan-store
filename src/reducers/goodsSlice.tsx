import { createSlice, PayloadAction } from '@reduxjs/toolkit'

type product = {
	url: string
	title: string
	description: string
	barcode: string
	price: number
	manufacturer: string
	brand: string
	careType: string[]
	type: string
	weight: string
	quantity: number
}

type goodsState = {
	goodsList: product[]
	basketList: product[]
	totalPrice: number
	totalQuantity: number
}

const initialState: goodsState = {
	goodsList: [],
	basketList: [],
	totalPrice: 0,
	totalQuantity: 0
}

const goodsSlice = createSlice({
	name: 'goods',
	initialState,
	reducers: {
		setGoods(state, action: PayloadAction<any>) {
			state.goodsList.push(action.payload)
		},
		addProduct(state, action: PayloadAction<any>) {
			state.goodsList.unshift(action.payload)
		},
		deleteProduct(state, action: PayloadAction<any>) {
			state.goodsList = state.goodsList.filter(item => item.barcode !== action.payload)
		},
		addToBasket(state, action: PayloadAction<any>) {
			let currentProduct = state.basketList.find(i => i.barcode === action.payload.barcode)
			if (currentProduct) {
				currentProduct.quantity += action.payload.quantity
			} else {
				state.basketList.push(action.payload)
			}
			state.totalPrice += action.payload.price * action.payload.quantity
			state.totalQuantity += action.payload.quantity
		},
		removeProduct(state, action: PayloadAction<any>) {
			state.basketList = state.basketList.filter(item => item.barcode !== action.payload.barcode)
			state.totalPrice -= action.payload.price * action.payload.quantity
			state.totalQuantity -= action.payload.quantity
		},
		increaseQuantity(state, action: PayloadAction<any>) {
			let currentProduct = state.basketList.find(i => i.barcode === action.payload)
			if (currentProduct) {
				currentProduct.quantity += 1
				state.totalPrice += currentProduct.price
				state.totalQuantity++
			}

		},
		decreaseQuantity(state, action: PayloadAction<any>) {
			let currentProduct = state.basketList.find(i => i.barcode === action.payload)
			if (currentProduct) {
				currentProduct.quantity -= 1
				state.totalQuantity--
				state.totalPrice -= currentProduct.price
				if (currentProduct.quantity === 0) {
					state.basketList = state.basketList.filter(item => item.barcode !== action.payload)
				}
			}
		},
		editCard(state, action) {
			let currentProduct = state.goodsList.find(i => i.barcode === action.payload.id)
			if (currentProduct) {
				currentProduct.barcode = action.payload.newBarcode
				currentProduct.brand = action.payload.brand
				currentProduct.title = action.payload.title
				currentProduct.price = action.payload.price
				currentProduct.manufacturer = action.payload.manufacturer
				currentProduct.careType = action.payload.selected
			}

		},
		clearBasket(state) {
			state.basketList = initialState.basketList
			state.totalPrice = initialState.totalPrice
			state.totalQuantity = initialState.totalQuantity
		}
	}

})

export const { setGoods, addProduct, deleteProduct, addToBasket, removeProduct, increaseQuantity, decreaseQuantity, clearBasket, editCard } = goodsSlice.actions

export default goodsSlice.reducer;