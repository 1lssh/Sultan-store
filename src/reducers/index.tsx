import goodsReducer from "./goodsSlice";
import { configureStore } from "@reduxjs/toolkit";



const store = configureStore({
	reducer: {
		goods: goodsReducer
	}
})

export default store

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatach = typeof store.dispatch;