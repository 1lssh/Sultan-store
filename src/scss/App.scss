@import '../scss/header.scss';
@import '../scss/catalog.scss';
@import '../scss/productCardPage.scss';
@import '../scss/basketPage.scss';
@import '../scss/footer.scss';

html {
	overflow-y: overlay;
}

body {
	font-family: 'Inter';
}

.container {
	max-width: 1350px;
	margin: 0 auto;
	padding: 0px 10px;
}

.link {
	text-decoration: none;
	color: black;
}


.row-full {
	background-color: #3F4E65;
	opacity: 0.2;
	width: 100vw;
	position: relative;
	margin-left: -50vw;
	height: 1px;
	left: 50%;
	margin-bottom: 20px;

}

.item-count {
	& button {
		padding: 4px 14px;
		background: linear-gradient(90deg, rgba(255, 198, 80, 0.3) 0%, rgba(254, 202, 110, 0.3) 97.25%);
		border-radius: 1000px;
		border: none;
	}

	& span {
		margin: 0px 15px;
	}
}

.price-list {
	&-sm {

		& button {
			padding: 12px 30px;
			font-weight: 700;
			font-size: 12px;
			line-height: 17px;
			color: #FFFFFF;
			background: #FFC85E;
			border-radius: 79px;
			border: none;
			cursor: pointer;
		}

		& img {
			margin-left: 13px;
		}
	}

	& button {
		padding: 20px 45px;
		font-weight: 700;
		font-size: 14px;
		line-height: 17px;
		color: #FFFFFF;
		background: #FFC85E;
		border-radius: 79px;
		border: none;
		cursor: pointer;
	}

	& img {
		margin-left: 15px;
	}
}

.vertical-line {
	width: 0px;
	height: 50px;
	border: 1px dashed #3F4E65;
	opacity: 0.1;

	@media screen and (max-width: 1042px) {
		display: none;
	}

	&-sm {
		width: 0px;
		height: 15px;
		border: 1px dashed #3F4E65;
		opacity: 0.1;
		margin: 0px 10px;
	}
}

.edit-confirm {
	margin-top: 20px;
	width: 30px;
}

.horizontal-line {
	width: 100%;
	height: 0;
	border: 1px dashed #3F4E65;
	opacity: 0.1;
	margin-bottom: 50px;
}

.breadcrumb {
	display: flex;
	align-items: center;
	margin-bottom: 30px;
	margin-top: 45px;

	@media screen and (max-width: 767px) {
		display: none;
	}

	&__item {
		font-weight: 300;
		color: #3F4E65;
		font-size: 14px;
	}
}

.search-input {
	max-width: 218px;
	display: flex;
	align-items: center;
	height: 60px;
	background-color: #EDEDED;
	border-radius: 36px;
	padding: 0px 10px;


	& input {
		max-width: 155px;
		padding: 10px;
		border: none;
		background-color: #EDEDED;

		&:focus {
			outline: none;
		}

		&::placeholder {
			font-family: 'Inter';
		}
	}

	& button {
		background: #FFC85E;
		border-radius: 1000px;
		border: none;
		padding: 10px 13px;
		cursor: pointer;
	}
}

.delete-btn {
	background-color: #FFC85E;
	border: none;
	border-radius: 100%;
	padding: 17px 19px;
	cursor: pointer;
}





//============================================================




//==================================================================



.pagination-container {
	display: flex;
	list-style-type: none;
	justify-content: center;

	@media screen and (max-width: 782px) {
		padding: 0;
	}

	.pagination-item {
		padding: 0 12px;
		height: 32px;
		text-align: center;
		margin: auto 4px;
		color: rgba(0, 0, 0, 0.87);
		display: flex;
		box-sizing: border-box;
		align-items: center;
		letter-spacing: 0.01071em;
		border-radius: 16px;
		line-height: 1.43;
		font-size: 13px;
		min-width: 32px;

		&.dots:hover {
			background-color: transparent;
			cursor: default;
		}

		&:hover {
			background-color: rgba(0, 0, 0, 0.04);
			cursor: pointer;
		}

		&.selected {
			background-color: #FFC6504D;
		}

		.arrow {
			&::before {
				position: relative;
				/* top: 3pt; Uncomment this to lower the icons as requested in comments*/
				content: '';
				/* By using an em scale, the arrows will size with the font */
				display: inline-block;
				width: 0.4em;
				height: 0.4em;
				border-right: 0.20em solid #FFC85E;
				border-top: 0.20em solid #FFC85E;
			}

			&.left {
				transform: rotate(-135deg) translate(-50%);
			}

			&.right {
				transform: rotate(45deg);
			}
		}

		&.disabled {
			pointer-events: none;
			opacity: 0.5;

			.arrow::before {
				border-right: 0.20em solid #FFC85E;
				border-top: 0.20em solid #FFC85E;
			}

			&:hover {
				background-color: transparent;
				cursor: default;
			}
		}
	}
}