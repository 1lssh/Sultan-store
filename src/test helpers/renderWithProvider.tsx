import { Provider } from "react-redux"
import { MemoryRouter } from "react-router-dom"
import store from "../reducers"
import { render } from "@testing-library/react"



export const renderWithProvider = (component: any, initialRoute = '/') => {
	return (
		render(
			<Provider store={store}>
				<MemoryRouter initialEntries={[initialRoute]}>
					{component}
				</MemoryRouter>
			</Provider>
		)

	)
}