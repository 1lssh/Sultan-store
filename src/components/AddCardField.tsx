import { FC, useState } from "react";
import { useAppDispatch } from "../hook";
import { addProduct } from "../reducers/goodsSlice";
import { MultiSelectDropdown } from "./MultiSelectdropdown";
interface IaddCardFieldProps {
	setAddFieldActive: any
};

export const AddCardField: FC<IaddCardFieldProps> = (props) => {
	const [url, setUrl] = useState('')
	const [title, setTitle] = useState('')
	const [type, setType] = useState('')
	const [weight, setWeight] = useState('')
	const [barcode, setBarcode] = useState('')
	const [manufacturer, setManufacturer] = useState('')
	const [brand, setBrand] = useState('')
	const [description, setDescription] = useState('')
	const [price, setPrice] = useState(0)

	const [selected, setSelected] = useState<string[]>([])



	const toggleOption = (option: string) => {
		setSelected(prevSelected => {
			const newArray = [...prevSelected]
			if (newArray.includes(option)) {
				return newArray.filter(item => item !== option)
			} else {
				newArray.push(option)
				return newArray;
			}
		})
	}

	const titleChange = (e: React.FormEvent<HTMLInputElement>) => {
		setTitle(e.currentTarget.value)
	}
	const urlChange = (e: React.FormEvent<HTMLInputElement>) => {
		setUrl(e.currentTarget.value)
	}
	const typeChange = (e: React.FormEvent<HTMLInputElement>) => {
		setType(e.currentTarget.value)
	}
	const weightChange = (e: React.FormEvent<HTMLInputElement>) => {
		setWeight(e.currentTarget.value)
	}
	const brandChange = (e: React.FormEvent<HTMLInputElement>) => {
		setBrand(e.currentTarget.value)
	}
	const barcodeChange = (e: React.FormEvent<HTMLInputElement>) => {
		setBarcode(e.currentTarget.value)
	}
	const descriptionChange = (e: React.FormEvent<HTMLInputElement>) => {
		setDescription(e.currentTarget.value)
	}
	const manufacturerChange = (e: React.FormEvent<HTMLInputElement>) => {
		setManufacturer(e.currentTarget.value)
	}
	const priceChange = (e: React.FormEvent<HTMLInputElement>) => {
		setPrice(+e.currentTarget.value)
	}

	const product = {
		url: url,
		title: title,
		barcode: barcode,
		price: price,
		manufacturer: manufacturer,
		careType: selected,
		quantity: 1
	}


	const dispatch = useAppDispatch()


	return (
		<div >
			<div data-testid='add-field-elem' className="add-field">
				<div >
					<div className='add-field__title' >Изображение</div>
					<input value={url} onChange={urlChange} type="text" />
				</div>
				<div>
					<div className='add-field__title' >Название</div>
					<input value={title} onChange={titleChange} type="text" />
				</div>
				<div>
					<div className='add-field__title' >Тип ухода</div><MultiSelectDropdown selected={selected} toggleOption={toggleOption} />
				</div>
				<div>
					<div className='add-field__title' >Тип</div>
					<input value={type} onChange={typeChange} type="text" />
				</div>
				<div>
					<div className='add-field__title' >Вес</div>
					<input value={weight} onChange={weightChange} type="text" />
				</div>
				<div>
					<div className='add-field__title' >Штрихкод</div>
					<input value={barcode} onChange={barcodeChange} type="text" />
				</div>

				<div>
					<div className='add-field__title' >Производитель</div>
					<input value={manufacturer} onChange={manufacturerChange} type="text" />
				</div>
				<div>
					<div className='add-field__title' >Бренд</div>
					<input value={brand} onChange={brandChange} type="text" />
				</div>

				<div>
				</div>
				<div>
					<div className='add-field__title' >Описание</div>
					<input value={description} onChange={descriptionChange} type="text" />
				</div>
				<div>
					<div className='add-field__title' >Цена</div>
					<input value={price} onChange={priceChange} type="number" />
				</div>
				<button onClick={() => dispatch(addProduct(product))}>Добавить</button>
				<button onClick={() => props.setAddFieldActive(false)}>Отмена</button>
			</div>

		</div>
	);


}