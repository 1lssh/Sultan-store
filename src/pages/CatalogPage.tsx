import { FC, useMemo, useState } from "react";
import { useAppSelector } from '../hook';
import { CareFilterOptions } from '../components/CareFilterOptions';
import { Sidebar } from '../components/Sidebar';
import { AddCardField } from '../components/AddCardField';
import { GoodsCard } from '../components/GoodsCard';
import Pagination from '../components/Pagination';
import { SortSelect } from '../components/SortSelect';


export const CatalogPage: FC = (props) => {
	let getGoods = useAppSelector(state => state.goods.goodsList)

	const [currentPage, setCurrentPage] = useState<number>(1);


	const [minPrice, setMinPrice] = useState<any>('')
	const [maxPrice, setMaxPrice] = useState<any>('')


	const [filterTags, setFilterTags] = useState<string[]>([])
	const [careFilterTags, setCareFilterTags] = useState<string[]>([])

	const [sortArrowActive, setSortArrowActive] = useState<boolean>(false)
	const [sortSelect, setSortSelect] = useState<string>('title')

	const [addFieldActive, setAddFieldActive] = useState<boolean>(false)


	const [careOptionsChecked, setCareOptionsChecked] = useState({
		handCare: false,
		bodyCare: false,
		faceCare: false,
		footCare: false,
		hairCare: false
	})


	const getCareFilterItem = (arr: any) => {
		let currentItem
		for (let i = 0; i < arr.length; i++) {
			if (careFilterTags.includes(arr[i])) {
				currentItem = arr[i]
				break;
			}
		}
		return currentItem
	}

	const filteredData = getGoods.filter((i) =>
		filterTags.length > 0
			? filterTags.includes(i.manufacturer)
			: getGoods
	).filter((i) =>
		careFilterTags.length > 0
			?
			careFilterTags.includes(getCareFilterItem(i.careType))
			: getGoods

	).filter(i => {
		if (maxPrice === '') return i.price >= minPrice
		if (minPrice === '') return i.price <= maxPrice
		return +i.price >= +minPrice && +i.price <= +maxPrice
	})


	const filterHandler = (e: React.FormEvent<HTMLInputElement>) => {
		if (e.currentTarget.checked) {
			setFilterTags([...filterTags, e.currentTarget.value])
		} else {
			setFilterTags(
				filterTags.filter((filterTag) => filterTag !== e.currentTarget.value)
			)
		}
	}

	const careFilterHandler = (e: React.FormEvent<HTMLInputElement>) => {
		if (e.currentTarget.checked) {
			setCareFilterTags([...careFilterTags, e.currentTarget.value])
		} else {
			setCareFilterTags(
				careFilterTags.filter((filterTag) => filterTag !== e.currentTarget.value)
			)
		}
		console.log(careFilterTags)
	}

	const sortHandler = (arr: any) => {
		if (sortSelect === 'title' && !sortArrowActive) {
			return arr.sort((x: any, y: any) => x.title.localeCompare(y.title))
		}
		if (sortSelect === 'title' && sortArrowActive) {
			return arr.sort((x: any, y: any) => x.title.localeCompare(y.title)).reverse()
		}
		if (sortSelect === 'price' && !sortArrowActive) {
			return arr.sort((x: any, y: any) => x.price - y.price)
		}
		if (sortSelect === 'price' && sortArrowActive) {
			return arr.sort((x: any, y: any) => y.price - x.price)
		}
		return arr
	}

	const onSelectChange = (e: React.FormEvent<HTMLSelectElement>) => {
		setSortSelect(e.currentTarget.value)
	}


	let PageSize = 15;

	const currentData = useMemo(() => {
		const firstPageIndex = (currentPage - 1) * PageSize;
		const lastPageIndex = firstPageIndex + PageSize;
		return sortHandler(filteredData).slice(firstPageIndex, lastPageIndex);
	}, [currentPage, sortHandler(filteredData)]);

	return (
		<div data-testid='catalog-page'>
			<div>
				<div className="breadcrumb">
					<div className='breadcrumb__item'>Главная</div>
					<div className='vertical-line-sm'></div>
					<div className='breadcrumb__item'>Косметика и гигиена</div>
				</div>
				<div className="catalog">
					<div className="catalog__head">
						<div className="catalog__title">
							<h1>Косметика и гигиена</h1>
						</div>
						<div className="catalog__desktop-sort">
							<SortSelect
								onSelectChange={onSelectChange}
								setSortArrowActive={setSortArrowActive}
								sortArrowActive={sortArrowActive}
							/>
						</div>
						<CareFilterOptions
							blockClassName='catalog__sort-options'
							itemClassName='catalog__sort-option'
							careFilterHandler={careFilterHandler}
							setCurrentPage={setCurrentPage}
							careOptionsChecked={careOptionsChecked}
							setCareOptionsChecked={setCareOptionsChecked}
						/>

					</div>
					<div className="catalog__main">
						<Sidebar
							goods={getGoods}
							filterHandler={filterHandler}
							careFilterHandler={careFilterHandler}
							setFilterTags={setFilterTags}
							minPrice={minPrice}
							setMinPrice={setMinPrice}
							maxPrice={maxPrice}
							setMaxPrice={setMaxPrice}
							setCurrentPage={setCurrentPage}
							setSortSelect={setSortSelect}
							sortArrowActive={sortArrowActive}
							setSortArrowActive={setSortArrowActive}
							careOptionsChecked={careOptionsChecked}
							setCareOptionsChecked={setCareOptionsChecked}
						/>

						<div className="catalog__goods">
							{
								addFieldActive ? <AddCardField setAddFieldActive={setAddFieldActive} />
									:
									<div className='add-btn'>
										<button data-testid='show-add-field-btn' onClick={() => setAddFieldActive(true)}>+</button>
									</div>
							}

							<div className="catalog__goods-grid">
								{
									currentData.map((item: any) => {
										return (
											<GoodsCard
												key={item.barcode}
												description={item.description}
												url={item.url}
												title={item.title}
												barcode={item.barcode}
												brand={item.brand}
												careType={item.careType}
												type={item.type}
												price={item.price}
												weight={item.weight}
												manufacturer={item.manufacturer} />
										)
									})
								}
							</div>

							<div className='catalog__pagination'>
								<Pagination
									className="pagination-bar"
									currentPage={currentPage}
									totalCount={sortHandler(filteredData).length}
									pageSize={PageSize}
									onPageChange={(page: any) => setCurrentPage(page)}
								/>
							</div>
							<div className="catalog__text">
								Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam interdum ut justo, vestibulum sagittis iaculis iaculis. Quis mattis vulputate feugiat massa vestibulum duis. Faucibus consectetur aliquet sed pellentesque consequat consectetur congue mauris venenatis. Nunc elit, dignissim sed nulla ullamcorper enim, malesuada.
							</div>
						</div>
					</div>
				</div>
			</div>
		</div>
	);
}
