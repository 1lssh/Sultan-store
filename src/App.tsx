import { useEffect } from 'react'
import './css/App.css';
import { Route, Routes } from 'react-router-dom';
import { ProductCardPage } from './pages/ProductCardPage';
import Layout from './components/Layout';
import { CatalogPage } from './pages/CatalogPage';
import { BasketPage } from './pages/BasketPage';
import { useAppDispatch, useAppSelector } from './hook';
import { setGoods } from './reducers/goodsSlice';
import goods from './goods.json'


function App() {
	let getGoods = useAppSelector(state => state.goods.goodsList)
	const dispatch = useAppDispatch()
	useEffect(() => {
		if (getGoods.length === 0) goods.forEach(i => dispatch(setGoods(i)))
	}, [getGoods])

	return (
		<>
			<Routes>
				<Route path='/' element={<Layout />}>
					<Route path=':id' element={<ProductCardPage />} />
					<Route index element={<CatalogPage />} />
					<Route path='/basket' element={<BasketPage />} />
				</Route>
			</Routes>
		</>

	);
}

export default App;
