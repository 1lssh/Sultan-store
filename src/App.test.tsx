import { fireEvent, screen } from '@testing-library/react';
import { ToBasketBtn } from './components/ToBasketBtn';
import App from './App';
import { renderWithProvider } from '../src/test helpers/renderWithProvider';
import * as actions from './reducers/goodsSlice'
import { CatalogPage } from './pages/CatalogPage';


describe('testing components', () => {
	test('add field should not to be in the document by default', () => {
		renderWithProvider(<CatalogPage />)

		const buttonElement = screen.queryByText(/добавить/i);
		expect(buttonElement).not.toBeInTheDocument();
	});

	test('add button should not to be in the document after clicking', () => {
		renderWithProvider(<CatalogPage />)

		const showAddFieldBtn = screen.getByTestId('show-add-field-btn')
		expect(showAddFieldBtn).toBeInTheDocument()
		fireEvent.click(showAddFieldBtn)
		expect(showAddFieldBtn).not.toBeInTheDocument()

	});

	test('to basket button snapshot', () => {
		renderWithProvider(<ToBasketBtn />)

		const buttonElement = screen.getByRole('button');
		expect(buttonElement).toMatchSnapshot();
	});

	test('should direct to basket page', () => {
		renderWithProvider(<App />, '/basket')
		expect(screen.getByTestId("basket-page")).toBeInTheDocument()
	});

	test('should direct to catalog page', () => {
		renderWithProvider(<App />)
		expect(screen.getByTestId("catalog-page")).toBeInTheDocument()
	});

	test('should direct to product card page', () => {
		renderWithProvider(<App />, '/4604049097558')
		expect(screen.getByTestId("product-card-page")).toBeInTheDocument()
	});

	test('cards length should be <= 15', () => {
		renderWithProvider(<CatalogPage />)
		const productCard = screen.getAllByTestId('product-card')
		expect(productCard.length).toBeLessThanOrEqual(15)
	});

	test('should increment and decrement quantity', () => {
		renderWithProvider(<App />, '/4604049097558')

		const incrementBtn = screen.getByTestId('increment')
		const decrementBtn = screen.getByTestId('decrement')
		const counter = screen.getByTestId('counter')
		expect(counter.textContent).toBe('1')
		fireEvent.click(decrementBtn)
		expect(counter.textContent).toBe('1')
		fireEvent.click(incrementBtn)
		expect(counter.textContent).toBe('2')
		fireEvent.click(decrementBtn)
		expect(counter.textContent).toBe('1')

	});

	test('should dispatch action', () => {
		const mockedAction = jest.spyOn(actions, 'deleteProduct')
		renderWithProvider(<CatalogPage />)
		fireEvent.click(screen.getAllByTestId('img')[0])
		expect(mockedAction).toHaveBeenCalledWith('4604049097558')
		expect(mockedAction).toHaveBeenCalledTimes(1)
	});


})




