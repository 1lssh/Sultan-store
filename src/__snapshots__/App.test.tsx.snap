// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing components to basket button snapshot 1`] = `
<button>
  В корзину 
  <img
    alt="basket"
    src="card-basket.png"
  />
</button>
`;
